gdal/frmts/nitf/nitffile.c

```diff
@@ -2216,7 +2217,77 @@ static char** NITFGenericMetadataReadTREInternal(char **papszMD,
                 (*pnMDSize) ++;
                 papszTmp[0] = NULL;
                 CPLFree(papszTmp);
+                const char* pszVal = strchr(papszMD[(*pnMDSize) - 1], '=') + 1;
+                
+				if (strcmp(pszType, "IEEE754") == 0)
+				{
+                    int len = strlen(pszVal);
```

Comment:
all the below is over complicated. Basically everywhere in GDAL assumes that native float C type is IEEE754. So you could simplify this as approximatively:
```
float f;
memcpy(&f, pszVal, 4);
CPL_MSBPTR32(&f);
```

Reply:
Made this change